src/variables/table.tsx

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { ArrayExt } from '@lumino/algorithm';
+
+import { ReactWidget } from '@jupyterlab/apputils';
+
+import React, { useEffect, useState } from 'react';
+
+import { VariablesModel } from './model';
+
+import { convertType } from '.';
+
+import { IDebugger } from '../tokens';
+
+import { CommandIDs } from '..';
+
+import { CommandRegistry } from '@lumino/commands';
+
+/**
+ * The body for Table Variables .
+ */
+export class VariablesBodyTable extends ReactWidget {
+  /**
+   * Instantiate a new Body for the Table Variables .
+   * @param model The model for the variables.
+   */
+  constructor(options: VariablesBodyTable.IOptions) {
+    super();
+    this._model = options.model;
+    this._commands = options.commands;
+    this.addClass('jp-DebuggerVariables-body');
+    this._model.changed.connect(this.updateScopes, this);
+  }
+
+  private updateScopes(self: VariablesModel) {
```

Comment:
How about replacing `self` by a more descriptive name?

Reply:
Also this method could be renamed to `_updateScopes` to follow the code style, and have a docstring.